frontend/model/contracts/identity.js

```diff
@@ -70,18 +99,44 @@ sbp('chelonia/defineContract', {
         for (const key in initialState) {
           Vue.set(state, key, initialState[key])
         }
+      },
+      async sideEffect ({ contractID, data }) {
+        // Lookup and save the username so that we can verify that it matches
+        const lookupResult = await sbp('namespace/lookup', data.attributes.username)
+        if (lookupResult !== contractID) {
+          console.error(`Mismatched username. The lookup result was ${lookupResult} instead of ${contractID}`)
```

Comment:
Seems like a big deal, should an exception be thrown to alert the user?

Reply:
I think alerts should be used sparingly, and probably not here. This check is expected to fail if the username ever changes, so in reality we'd need to check for the latest username before notifying the user, and then I'd use a notification instead of throwing.